File: backend/Models/CriteriaTemplate.cs
namespace InternshipManagement.Api.Models;

public class CriteriaTemplate
{
    public Guid Id { get; set; }

    public Guid PeriodId { get; set; }

    public string Code { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public int Version { get; set; } = 1;

    public string Status { get; set; } = "DRAFT";

    public int MinItems { get; set; } = 3;

    public int MaxItems { get; set; } = 8;

    public bool AllowWeightEdit { get; set; } = true;

    public Guid? IssuedBy { get; set; }

    public DateTime? IssuedAt { get; set; }

    public InternshipPeriod Period { get; set; } = null!;

    public User? IssuedByUser { get; set; }

    public ICollection<CriteriaTemplateItem> Items { get; set; }
        = new List<CriteriaTemplateItem>();
}

File: frontend/src/types/criteriaTemplate.ts
export type CriteriaTemplateStatus =
  | "DRAFT"
  | "PUBLISHED";

export interface CriteriaTemplateItemResponse {
  id: string;
  name: string;
  defaultWeight: number;
  isMandatory: boolean;
  minPassScore: number | null;
  displayOrder: number;
}

export interface CriteriaTemplateResponse {
  id: string;
  periodId: string;
  periodCode: string;
  code: string;
  name: string;
  version: number;
  status: CriteriaTemplateStatus;
  minItems: number;
  maxItems: number;
  allowWeightEdit: boolean;
  weightTotal: number;
  issuedBy: string | null;
  issuedAt: string | null;
  items: CriteriaTemplateItemResponse[];
}

export interface CreateCriteriaTemplateRequest {
  periodId: string;
  code: string;
  name: string;
  minItems: number;
  maxItems: number;
  allowWeightEdit: boolean;
}

export interface AddCriteriaTemplateItemRequest {
  name: string;
  defaultWeight: number;
  isMandatory: boolean;
  minPassScore: number | null;
  displayOrder: number;
}
